File: apple/Sources/CCDAUI/Views/CCDATheme.swift
import SwiftUI

#if os(macOS)
import AppKit
#endif

/// Default visual tokens used by CCDAUI views.
public enum CCDATheme {
    /// Main clinical accent color.
    public static let primary = Color(red: 0.10, green: 0.43, blue: 0.48)
    /// Secondary accent color used for entry metadata.
    public static let secondary = Color(red: 0.28, green: 0.31, blue: 0.62)
    /// Warm accent color used for attachments.
    public static let attachment = Color(red: 0.74, green: 0.37, blue: 0.12)
    /// Soft page background.
    public static var pageBackground: Color {
        #if os(iOS)
        Color(.systemGroupedBackground)
        #elseif os(macOS)
        Color(nsColor: .windowBackgroundColor)
        #else
        Color.gray.opacity(0.08)
        #endif
    }
    /// Primary panel background.
    public static var panelBackground: Color {
        #if os(iOS)
        Color(.secondarySystemGroupedBackground)
        #elseif os(macOS)
        Color(nsColor: .controlBackgroundColor)
        #else
        Color.white
        #endif
    }
    /// Nested row background.
    public static var rowBackground: Color {
        #if os(iOS)
        Color(.tertiarySystemGroupedBackground)
        #elseif os(macOS)
        Color(nsColor: .textBackgroundColor).opacity(0.72)
        #else
        Color.gray.opacity(0.08)
        #endif
    }
    /// Card corner radius.
    public static let cornerRadius: CGFloat = 8
}

File: android/ccda-compose/src/main/kotlin/com/ccdakit/compose/CCDATheme.kt
package com.ccdakit.compose

import androidx.compose.ui.graphics.Color
import androidx.compose.ui.unit.dp

/** Default visual tokens used by ccda-compose views. */
object CCDATheme {
    /** Main clinical accent color. */
    val Primary = Color(0xFF1A6D7A)
    /** Secondary accent color used for entry metadata. */
    val Secondary = Color(0xFF474F9E)
    /** Warm accent color used for attachments. */
    val Attachment = Color(0xFFBD5E1F)
    /** Soft page background. */
    val PageBackground = Color(0xFFF3F6F7)
    /** Dark-mode page background. */
    val DarkPageBackground = Color(0xFF101719)
    /** Card corner radius. */
    val CornerRadius = 8.dp
}
